src/qt/transactionview.cpp

```diff
@@ -37,6 +39,37 @@
 #include <QUrl>
 #include <QVBoxLayout>
 
+class TransactionRecordDelegate : public QItemDelegate
+{
+    QSortFilterProxyModel* m_proxy;
+
+    void paint(QPainter* painter, const QStyleOptionViewItem& option, const QModelIndex& index) const
+    {
+        bool alternate = false;
+        QVariant previous_hash;
+
+        for (int row = 0; row <= index.row(); ++row) {
+            QModelIndex sibling = m_proxy->mapToSource(index.sibling(row, 0));
+            QVariant hash = sibling.data(TransactionTableModel::TxHashRole);
+            if (row == 0) {
+                previous_hash = hash;
+            } else if (hash != previous_hash) {
+                alternate = !alternate;
+                previous_hash = hash;
+            }
+        }
+
+        if (alternate) {
+            painter->fillRect(option.rect, option.palette.alternateBase());
+        }
+
+        QItemDelegate::paint(painter, option, index);
+    }
+
+public:
+    TransactionRecordDelegate(QSortFilterProxyModel* proxy) : m_proxy(proxy) {}
```

Comment:
```
2018-09-19 03:32:23 clang-tidy(pr=12578): src/qt/transactionview.cpp:70:5: warning: single-argument constructors must be marked explicit to avoid unintentional implicit conversions [google-explicit-constructor]
```